# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../openlibraries/matching-function.patch
# Copyright (C) 2007 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---


  Original patch from http://openlibraries.cvs.sourceforge.net/openlibraries/openlibraries/src/openobjectlib/sg/appearance.cpp?r1=1.18&r2=1.19&view=patch
  but slightly adjusted.

--- a/src/openobjectlib/sg/appearance.cpp	2007/05/17 18:48:19	1.18
+++ b/src/openobjectlib/sg/appearance.cpp	2007/07/20 10:16:31	1.19
@@ -53,12 +53,15 @@
 		
 		opl::wstring filename_;
 	};
-	
+
 	bool find_url_availability( const opl::m_url& m, fs::path& p )
 	{
 		typedef opl::m_url::const_iterator const_iterator;
-		
-		const_iterator I = std::find_if( m.begin( ), m.end( ), fs::exists );
+
+		//M> for some reason it MSVC8 doesnt like this call, and needs help with the template types
+		typedef bool (*fileSystemFunc)( const boost::filesystem::path & ph );
+
+		const_iterator I = std::find_if<opl::m_url::const_iterator, fileSystemFunc>(m.begin() , m.end(), fs::exists );
 		if( I != m.end( ) )
 		{
 			p = *I;
